code/modules/vore/eating/bellymodes_vr.dm

```diff
@@ -105,259 +107,195 @@
-									if(L && L.client && L.is_preference_enabled(/datum/client_preference/digestion_noises))
-										SEND_SOUND(L,prey_digest)
-									play_sound = pick(pred_digest)
+								if(IM_DIGEST_FOOD)
+									if(istype(I,/obj/item/weapon/reagent_containers/food) || istype(I,/obj/item/organ))
+										digest_item(I)
+									else
+										items_preserved |= I
+									if(prob(25)) //Less often than with normal digestion
+										if(H && H.client && H.is_preference_enabled(/datum/client_preference/digestion_noises))
+											SEND_SOUND(H, prey_digest)
+										play_sound = pick(pred_digest)
+								if(IM_DIGEST)
+									if(!H && (I.digest_stage && I.digest_stage > 0))
+										digest_item(I)
+									else
+										digest_item(I)
+										did_an_item = TRUE
+									if(prob(25)) //Less often than with normal digestion
+										if(H && H.client && H.is_preference_enabled(/datum/client_preference/digestion_noises))
+											SEND_SOUND(H, prey_digest)
+										play_sound = pick(pred_digest)
 							to_update = TRUE
-							break
+							
 		//get rid of things like blood drops and gibs that end up in there
-		else if(istype(A,/obj/effect/decal/cleanable/))
+		else if(istype(A, /obj/effect/decal/cleanable))
 			qdel(A)
 
-///////////////////////////// DM_HOLD /////////////////////////////
-	if(digest_mode == DM_HOLD)
-		return SSBELLIES_PROCESSED //Pretty boring, huh
-
-//////////////////////////// DM_DIGEST ////////////////////////////
-	else if(digest_mode == DM_DIGEST)
-
-		if(prob(50)) //Was SO OFTEN. AAAA.
-			for(var/mob/M in contents)
-				if(M && M.client && M.is_preference_enabled(/datum/client_preference/digestion_noises))
-					SEND_SOUND(M,prey_digest)
-			play_sound = pick(pred_digest)
-
-		for (var/target in touchable_mobs)
-			var/mob/living/M = target
-			//Pref protection!
-			if (!M.digestable || M.absorbed)
-				continue
-
-			//Person just died in guts!
-			if(M.stat == DEAD)
-				var/digest_alert_owner = pick(digest_messages_owner)
-				var/digest_alert_prey = pick(digest_messages_prey)
-				var/compensation = M.getOxyLoss() //How much of the prey's damage was caused by passive crit oxyloss to compensate the lost nutrition.
-
-				//Replace placeholder vars
-				digest_alert_owner = replacetext(digest_alert_owner,"%pred",owner)
-				digest_alert_owner = replacetext(digest_alert_owner,"%prey",M)
-				digest_alert_owner = replacetext(digest_alert_owner,"%belly",lowertext(name))
-
-				digest_alert_prey = replacetext(digest_alert_prey,"%pred",owner)
-				digest_alert_prey = replacetext(digest_alert_prey,"%prey",M)
-				digest_alert_prey = replacetext(digest_alert_prey,"%belly",lowertext(name))
-
-				//Send messages
-				to_chat(owner,"<span class='notice'>" + digest_alert_owner + "</span>")
-				to_chat(M,"<span class='notice'>" + digest_alert_prey + "</span>")
+	if(digest_mode == DM_TRANSFORM)
+		process_tf(tf_mode, touchable_mobs)
+		return SSBELLIES_PROCESSED
```

Comment:
Returning early here skips the code at the bottom of the switch statement for play_sound or updating vore panel (the flags for which may have been set above)